docs/architecture/adr-006-hermes-v0.2-usecases.md

```diff
@@ -0,0 +1,203 @@
+    - This pattern should also include a retrying mechanism.
+
+### Concrete Use-Cases
+
+Applying the above patterns to a few cases, we get the following concrete
+commands that Hermes v0.2.0 should fulfil.
+
+#### Create New Connection
+
+- Minimal invocation: this will create the connection using _new_ clients:
+
+```
+hermes create connection ibc-0 ibc-1 [--delay <delay>]
+```
+
+**Details:**
+Starts a transaction to perform the connection open handshake protocol between
+chains `ibc-0` and `ibc-1`. The optional parameter `--delay` is the delay period
+that the new connection should have.
+
+- Reusing pre-existing state, concretely, with _existing_ clients:
+
+```
+hermes create connection ibc-0 --src-client-id <client-id> --dst-client-id <client-id> [--delay <delay>]
+```
+
+**Details:**
+Similar to the previous command, but will reuse the client with identifier from
+option `--src-client-id` which is expected to exist on chain `ibc-0`. The
+[client state][client-state] from this client will also provide the identifier 
+for the destination chain (this is the "chain_id" field from the client state).
+Similarly, on the destination chain, this command will establish the connection
+using the client with identifier from the option `--dst-client-id`, which must
+be verifying headers for the source chain.
+
+##### Create New Channel
+
+- With _new_ connection and clients:
+
+```
+hermes create channel ibc-0 ibc-1 --src-port <port-id> --dst-port <port-id> --order <order> --version <version>
+```
+
+- With _existing_ specific connection:
+
+```
+hermes create channel ibc-0 --src-connection <connection-id> --src-port <port-id> --dst-port <port-id> --order <order> --version <version>
+```
+
+##### Packet Relaying
+
+- relay packets over a _new_ channel, _new_ connection, and _new_ clients:
+
+```
+hermes start ibc-0 ibc-1 --src-port <port-id> --dst-port <port-id>
+```
+
+- relay packets over an _existing_ channel:
+
+```
+hermes start ibc-0 --src-channel <channel-id>
+```
+
+- relay packets over a _new_ channel that uses an _existing_ connection:
+
+```
+hermes start ibc-0 --src-connection <connection-id> --src-port <port-id> --dst-port <port-id> --order <order>
+--version <version>
+```
+
+- relay packets over a _new_ channel that uses a _new_ connection that builds
+  on an _existing_ client:
+
+```
+hermes start ibc-0 --src-client-id <client-id> --dst-client-id <client-id> --delay <delay>
+```
+
+##### Finishing partially complete handshakes:
+
+These commands serve the purpose of covering certain corner-cases where a
+handshake may be partially started.
+
+- Finalize handshake for _partially established_ connection:
+
+```
+hermes create connection ibc-0 --src-connection <connection-id>
+```
+
+- Finalize handshake for _partially established_ channel:
+
+```
+hermes create channel ibc-0 --src-channel <channel-id>
+```
+
+
+### Command Output
+
+By default, we will provide human-readable output, i.e., pretty printing.
```

Comment:
Can we try to avoid "we" and "will"? e.g. "By default, a human-readable output is provided"...or "the commands have human-readable output..."
Assume you look at this once it is approved and implemented.

Reply:
Got it!
cb2f5fcd